### ISSUE TITLE
What is the relationship with libvterm on LaunchPad?

### ISSUE THREAD

**Issue by yurivict** [TragicWarrior/libvterm#171]
This project https://launchpad.net/libvterm has version 0.1.4 (http://www.leonerd.org.uk/code/libvterm/).

**Comment by TragicWarrior** [comment]
@yurivict , I am not familiar with that library.  this libvterm has a long heritage and was formerly known as libRote.

**Comment by sskras** [comment]
So these are two different projects.  They are under different licenses: LGPL and MIT.
The name collision is a bit unfortunate.

@TragicWarrior:

(1) any ideas on why was libROTE forked?
(2) when did that happen, please?

The only reference to it managed to find is this post circa December 2007:
https://www.brain-dump.org/blog/integrating-utf-8-support-in-librote-for-use-by-dvtm/

The last its' commit says [`2007-12-21`](https://repo.or.cz/librote.git/commit/13d347a).
And the first commit of the other project under the same name says [`2007-11-20`](https://bazaar.launchpad.net/~libvterm/libvterm/trunk/revision/1).

So that same named project seems to be at least 1 month older than libROTE.

Sadly this repo seems to only have history since [`2015-06-23`](a7de95c), when it says it was migrated from Google Code.
So it's hard to be sure about age this fork.

**Comment by dimitry-ishenko** [comment]
The name choice is very tragic 😉, considering that the original [libvterm](https://www.leonerd.org.uk/code/libvterm/) is used by [emacs](https://github.com/akermu/emacs-libvterm) and [neovim](https://github.com/neovim/libvterm), and is included in basically every Linux distro (eg, [Arch](https://archlinux.org/packages/extra/x86_64/libvterm/), [Gentoo](https://packages.gentoo.org/packages/dev-libs/libvterm), [Ubuntu](https://packages.ubuntu.com/search?keywords=libvterm0), [Debian](https://packages.debian.org/search?searchon=names&keywords=libvterm0), [Alpine](https://pkgs.alpinelinux.org/packages?name=libvterm), [Fedora](https://packages.fedoraproject.org/pkgs/libvterm/libvterm/), etc).

**Comment by sskras** [comment]
Finally I discovered the [source of truth](https://code.google.com/archive/p/libvterm/source/default/commits) regarding the project start:

Author | Date | Commit | Message
-- | -- | -- | --
bryan.christ@gmail.com | Jun 17, 2013 | 5 | fixed the backspace key for getline() and other tools that modify terminal attri
bryan.christ@gmail.com | Jun 15, 2013 | 4 | added a test program for key strokes
bryan.christ@gmail.com | May 7, 2013 | 3 | updated the CHANGELOG
bryan.christ@gmail.com | May 4, 2013 | 2 | major update to escape code parser. added pathways to handle xterm osc sequence
  | May 4, 2013 | 1 | Initial directory structure.

The SVN-source archive seems to be also available: 
https://storage.googleapis.com/google-code-archive-source/v2/code.google.com/libvterm/source-archive.zip

It looks to contain only 4 commits circa 2013, which were forked into the current repo circa 2015.

Argh! Here seems to be another part of the SVN history embracing 2010-04-24 ... 2013-05-04 time range:
https://sourceforge.net/p/libvterm/code/commit_browser

So it should be still possible to recreate fuller history of the project (for those interested).

**Comment by sskras** [comment]
@TragicWarrior, so what do you think about adjusting the library name (at least a bit)?

Now that I asked you, the `libvtermROTE` came to my mind. :) But it looks a bit longish.

Also I tried to decipher the letter `R` in ROTE to no avail. Even the original SF page has no mention:
https://rote.sourceforge.net/

**Comment by sskras** [comment]
So I took opportunity to put it a dual way:

> **R**oll/ou**R** **O**wn **T**erminal **E**mulation

Now that the "Terminal" part is already abbreviated in current library name as `term`, and the "Emulation" became `v` as in "virtual", both **T** and **E** might be skipped leaving us with the two original letters: **R** and **O**.

As `libvtermRO` looks yet messy to me (still too long, unpronouncable, and ambiguous as in collision with "Read-Only") + because the letter **O** is present in all three words that's left (Roll/ouR Own), I guess it would be fair to take **only this letter** and add it to the current name. 

So this would become `libvtermo` – still readable, only marginally different from the current form, and possibly interpreted as "libvterm original" or maybe just "our own".

@TragicWarrior, would you mind such naming change? :)

**Comment by sskras** [comment]
PS. I mean, name adjustment would facilitate the library's spread in the world of open source.

At least I would try to promote it to the [**pkgsrc**](https://www.pkgsrc.org/) tree.
It already packages the colliding lib from _leonerd.org.uk_ under these names:
http://ftp.pkgsrc.org/pub/pkgsrc/current/pkgsrc/devel/libvterm01/index.html
http://ftp.pkgsrc.org/pub/pkgsrc/current/pkgsrc/devel/libvterm03/index.html

**Comment by TragicWarrior** [comment]
@sskras , i agree that it's unfortunate naming.  libvterm goes back to circa 2006 (on SourceForge).  then, I took over as the maintainer of libROTE (with the authors permission) which became the new libvterm as a fork.

I open to suggestions on the name change if that fosters adoption and attention on the project.  I would also agree that libvtermROTE is too long. LOL.

**Comment by sskras** [comment]
Oh, cool. I presume you didn't like `libvtermo` as the replacement too, right? :)
